src/components/ProductCreateEditPage/ProductCreateEditPage.tsx

```diff
@@ -0,0 +1,207 @@
+import { ActionGroup, Button, Form, FormGroup, FormHelperText, Label, TextArea, TextInput } from '@patternfly/react-core';
+import { useEffect } from 'react';
+import { useNavigate, useParams } from 'react-router-dom';
+
+import { productEntityAttributes } from 'common/productEntityAttributes';
+
+import { IFieldValues, useForm } from 'hooks/useForm';
+import { useServiceContainer } from 'hooks/useServiceContainer';
+import { useTitle } from 'hooks/useTitle';
+
+import { ContentBox } from 'components/ContentBox/ContentBox';
+import { PageLayout } from 'components/PageLayout/PageLayout';
+import { ServiceContainerCreatingUpdating } from 'components/ServiceContainers/ServiceContainerCreatingUpdating';
+
+import * as productApi from 'services/productApi';
+
+import { createSafePatch } from 'utils/patchHelper';
+import { generatePageTitle } from 'utils/titleHelper';
+
+const formConfig = {
+  name: {
+    isRequired: true,
+  },
+  abbreviation: {
+    isRequired: true,
+  },
+};
+
+interface IProductCreateEditPageProps {
+  isEditPage?: boolean;
+}
+
+export const ProductCreateEditPage = ({ isEditPage = false }: IProductCreateEditPageProps) => {
+  const { productId } = useParams();
+  const navigate = useNavigate();
+
+  // create page
+  const serviceContainerCreatePage = useServiceContainer(productApi.createProduct);
+
+  // edit page - get method
+  const serviceContainerEditPageGet = useServiceContainer(productApi.getProduct);
+  const serviceContainerEditPageGetRunner = serviceContainerEditPageGet.run;
+
+  // edit page - patch method
+  const serviceContainerEditPagePatch = useServiceContainer(productApi.patchProduct);
+
+  const { register, setFieldValues, getFieldState, getFieldErrors, handleSubmit, isSubmitDisabled } = useForm();
+
+  useTitle(
+    generatePageTitle({
+      pageType: isEditPage ? 'Edit' : 'Create',
+      serviceContainer: serviceContainerEditPageGet,
+      firstLevelEntity: 'Product',
+    })
+  );
+
+  const submitCreate = (data: IFieldValues) => {
+    return serviceContainerCreatePage
+      .run({
+        serviceData: { data },
+      })
+      .then((response: any) => {
+        const newProductId = response?.data?.id;
+        if (!newProductId) {
+          throw new Error(`Invalid productId coming from Orch POST response: ${newProductId}`);
+        }
+        navigate(`/products/${newProductId}`);
+      })
+      .catch((e: any) => {
+        throw new Error('Failed to create Product.');
```

Comment:
Can you please make sure if errors like this are handled properly? Feel free to create new jira and do some additonal investigation. We need to be sure:
1) user knows what he is supposed to do when error happends
2) all errors which should be logged are logged

Reply:
I created new jiras NCL-8169 and NCL-8170